### ISSUE TITLE
NPE when using a MockResourceAccessor with contents being an XML string

### ISSUE THREAD

**Issue by carlocarlen** [liquibase/liquibase#2125]
## Environment

**Liquibase Version**: 4.5.0

**Liquibase Integration & Version**: gradle

**Liquibase Extension(s) & Version**: liquibase-core 4.5.0

**Database Vendor & Version**: MariaDB 10.3

**Operating System Type & Version**: Windows 10

## Description

A Null Pointer Exception is thrown when using a MockResourceAccessor with contents being an XML string.
NPE is thrown by LiquibaseEntityResolver.resolveEntity(LiquibaseEntityResolver.java:40)
Full StackTrace is attached (stackTrace.txt)

## Steps To Reproduce

Create a MockResourceAccessor(contentByFileName) with contentByFileName being an XML string generated in the code.
Run the update.
A simplified code extract is attached (PseudoCode.java)

## Actual Behavior
Null Pointer Exception, see stackTrace.txt

## Expected/Desired Behavior
No Null Pointer Exception

## Additional Information
By debugging the code, I believe this occurs when MockResourceAccessor tries to load the xsd schema, it does not find it and returns null.
Then, LiquibaseEntityResolver should search for it using a "FallBackResourceAccessor". However, since the MockResourceAccessor returned null instead of an empty streams object, a NPE is thrown.

Working on a project where both a lot of xml scripts and dynamically generated scripts are used, I was able to dig a bit deeper in this problem.
The present issue is similar to an issue addressed by this PR: https://github.com/liquibase/liquibase/pull/1621
However that PR introduced a new issue:  https://github.com/liquibase/liquibase/issues/1979
Solved by PR https://github.com/liquibase/liquibase/pull/1860 which is merged in 4.5.0
I believe this last PR reintroduced the current bug.

## Attachments: 

[PseudoCode.java.txt](https://github.com/liquibase/liquibase/files/7250190/PseudoCode.java.txt)
[stackTrace.txt](https://github.com/liquibase/liquibase/files/7250191/stackTrace.txt)

**Comment by molivasdat** [comment]
Hi @carlocarlen Thanks for alerting us to this issue.  We will add this to the list of issues to process.

**Comment by tati-qalified** [comment]
Hello @carlocarlen , sorry for the late response, we're trying to catch up and review older issues such as this one.
I tested this with the latest Liquibase 4.23.0 version, and it seems to work just fine. I tried running two different changelogs just to be sure, and both ran successfully. 

I'll be closing this issue, though be sure to let us know if you have any questions or if the problem persists.
Thanks,
Tatiana
